Ge 130

bgschiller opened the pull request:
sorry for the giant PR, @devKate.

I'm going to branch off here to make the modifications NREL suggested today, but this is stable and tests are passing. So it can be merged when you're ready.

gh-23378 committed:
reworking toolbar layout

gh-23378 committed:
menus

gh-23378 committed:
stub

gh-23378 committed:
loading svgs

gh-23378 committed:
include svgs as components

bgschiller committed:
bit of progress on editable select

gh-23378 committed:
more css

bgschiller committed:
use inputs always

bgschiller committed:
Merge branch 'develop' into editable-select

bgschiller committed:
delete rows too

bgschiller committed:
updates working

bgschiller committed:
styling changes

bgschiller committed:
refactor select

bgschiller committed:
refactor validated input

bgschiller committed:
working dbl-click to focus, but seems fragile

bgschiller committed:
maybe this will work?

gh-23378 committed:
component tabs

gh-23378 committed:
faked the tabs with CSS

bgschiller committed:
Finish up select, edit on dblclick, etc

bgschiller committed:
Merge branch 'develop' into GE-130

bgschiller committed:
Merge branch 'GE-130' into editable-select

bgschiller committed:
fix path

gh-23378 committed:
components

gh-23378 committed:
Merge branch 'GE-130' of https://github.com/NREL/openstudio-geometry-editor into GE-130

bgschiller committed:
katie's recommendations

bgschiller committed:
Merge branch 'GE-130' into editable-select

bgschiller committed:
Merge pull request #223 from NREL/editable-select

Editable select

gh-23378 committed:
placement of editable select

bgschiller committed:
fix toolbar too big

bgschiller committed:
move svgs to one package

bgschiller committed:
mode-tabs working

bgschiller committed:
Fix svg choice

bgschiller committed:
visual changes to editable select

bgschiller committed:
move some buttons around in preparation for >> and << buttons

bgschiller committed:
library-goes-here working correctly

bgschiller committed:
stick the library in there

bgschiller committed:
library instead of nav list

bgschiller committed:
split library into presentation comp

bgschiller committed:
Factor select list into component

bgschiller committed:
library refactor coming together

bgschiller committed:
some more handlers hooked up

bgschiller committed:
expand/contract working

bgschiller committed:
css tweaks

bgschiller committed:
add search

bgschiller committed:
fix widths

bgschiller committed:
Got some more table stuff working

bgschiller committed:
add image upload component

bgschiller committed:
cut down toolbar size a bit

bgschiller committed:
Add row sorting

bgschiller committed:
visual changes to modal

bgschiller committed:
add key to fix weirdness with colors

bgschiller committed:
add image tools

bgschiller committed:
render images to svg, not canvas

bgschiller committed:
add controls

bgschiller committed:
move images with d3

bgschiller committed:
rotation coming along, but interferes with movement

bgschiller committed:
correct for extant rotation

bgschiller committed:
remove dead markers

bgschiller committed:
choose the best cursor for the job

bgschiller committed:
add rotation cursor

bgschiller committed:
Allow non-movements

bgschiller committed:
factor out drawMethods parameters

bgschiller committed:
remove canvas images

bgschiller committed:
remove konva from deps

(and add d3-selection, which I should have done back when I started
using it. But we had it anyway as a transitive dependency)

bgschiller committed:
Merge branch 'images-in-svg' into GE-130

bgschiller committed:
props down, events up. Update selected object when rows changes

bgschiller committed:
remove special images mode

bgschiller committed:
position component menu correctly

bgschiller committed:
position component dropdown

bgschiller committed:
separate components from library expanded

bgschiller committed:
Add image tool (to prevent moving image when trying to pan)

bgschiller committed:
turn on/off events according to tool

bgschiller committed:
remove cursor: pointer when image is not clickable

bgschiller committed:
allow deselecting images

bgschiller committed:
Raise image when image tool is active

bgschiller committed:
Use svg double arrows, instead of ascii art

bgschiller committed:
replace inline svg with red delete icon

bgschiller committed:
you can't have 100% if your siblings are taking some space too

bgschiller committed:
widen story list

bgschiller committed:
fix some styles

bgschiller committed:
Place image at center of visible grid

bgschiller committed:
put back the optimization on zoom, pan

bgschiller committed:
avoid name clash mode/modeTab

bgschiller committed:
separate mode, modeTab and subselectionType

bgschiller committed:
Add AssignProperties menus

bgschiller committed:
remove watch

bgschiller committed:
remove console log

bgschiller committed:
switch to image tool when images are the subselection

bgschiller committed:
switch away from image tool when subselect changes

bgschiller committed:
avoid raising polygons, as they interfere with clicking grid to draw

bgschiller committed:
Switch to correct tool for Components, Assign

bgschiller committed:
fix deletion of component defs

bgschiller committed:
Merge branch 'develop' into GE-130

bgschiller committed:
fill in guideline text

bgschiller committed:
allow setting a 'current component instance

bgschiller committed:
windows can be deleted and modified

bgschiller committed:
window needs a name...

bgschiller committed:
highlight selected component

bgschiller committed:
only highlight in components mode tab

bgschiller committed:
replace perp-linecap with more flexible solution

bgschiller committed:
highlight dcs with same styles when selected

bgschiller committed:
remove component tool

bgschiller committed:
color selected dropdown on components list

bgschiller committed:
color selected dropdown on assign list

bgschiller committed:
Rip out dead code

bgschiller committed:
remove currentThermalZone,SpaceType,BuildingUnit in favor of SpaceProperty

bgschiller committed:
set current mode on space prop change

bgschiller committed:
reset mode when modeTab changes

bgschiller committed:
assign upon click

bgschiller committed:
use cursor:pointer on spaces when apply property tool is active

bgschiller committed:
components can only be added to spaces, not shading

bgschiller committed:
update colors working again

bgschiller committed:
fix pretty-select usage

bgschiller committed:
Merge branch 'develop' into GE-130

bgschiller committed:
fix huebee styles

bgschiller committed:
fix point-path origin issues

bgschiller committed:
Add defaults to factory

bgschiller committed:
Add color to pitched roof

bgschiller committed:
improve padding, size on location search input

bgschiller committed:
add titles for tools, components, assignable props

bgschiller committed:
adjust spacing in library controls

bgschiller committed:
right-justify numeric inputs

bgschiller committed:
add pitched roofs in a buncha places

bgschiller committed:
let the tools breathe

bgschiller committed:
Add fill tool

bgschiller committed:
Adjust rotation tool size and appearance

bgschiller committed:
space for search, even if we're not using it

bgschiller committed:
set min widths for tables, adjust destroy sizing

bgschiller committed:
distinguish between neutral and shading

bgschiller committed:
shading should auto-create for each story

bgschiller committed:
zoom in/out by clicking

bgschiller committed:
push map over

bgschiller committed:
fix longstanding broken map bug

bgschiller committed:
zoom to work in progress

bgschiller committed:
ho lee crap

bgschiller committed:
clear startResolution on resolveBounds

bgschiller committed:
fix a few test selectors

bgschiller committed:
all daylighting control tests working

bgschiller committed:
window tests passing

bgschiller committed:
fix remaining e2e tests

bgschiller committed:
add config option for snapMode

bgschiller committed:
clicking image tool should select an image

bgschiller committed:
title for component dropdowns

bgschiller committed:
add name of currently selected component and property

gh-23378 commented on src/components/Grid/methods.js:
```diff
@@ -1114,17 +1201,37 @@ export default {
     svg
       .on('mousemove', this.handleMouseMove)
       .on('click', this.gridClicked);
+    window.d3 = d3;
```
do we want this here?

bgschiller committed:
components, assign expanded by default. selectable as tables

bgschiller committed:
assign should come before components

bgschiller committed:
limit height of component, assignable prop dropdowns

bgschiller committed:
allow searching for assignable props

bgschiller committed:
fix overlay

bgschiller committed:
indicate component that will be deleted with Remove Component

bgschiller committed:
merge floorplan and shading modetabs

bgschiller committed:
new icons

bgschiller committed:
adjust size of icons to match

bgschiller committed:
tweak image icon

bgschiller committed:
add total count to window defn and daylighting control defn tables

bgschiller committed:
magic reappearing sorting icons

bgschiller committed:
zoom to fit almost done

bgschiller committed:
fix up zoom to fit

bgschiller committed:
table scrolling mostly working

bgschiller committed:
replace color scheme

bgschiller committed:
fix broken test

bgschiller committed:
table scrolls correctly

bgschiller committed:
clear up weird color region on the right

https://trello-attachments.s3.amazonaws.com/58d428743111af1d0a20cf28/59ea4bfce16a8472fcef7e8d/175655522add37e10f13ac82d4bb4fd3/capture.png

bgschiller committed:
omg zoomTo is working!!

bgschiller committed:
Merge branch 'zoomTo-wip' into GE-130-tweaks

bgschiller committed:
fix cases I broke with that last fix

bgschiller committed:
expanded dropdown takes up whole width

bgschiller committed:
Merge branch 'develop' into GE-130

bgschiller committed:
add new story and space attributes, from #229

bgschiller committed:
scroll to the left, now scroll to the bottom

bgschiller committed:
Fix assign and component menus

bgschiller committed:
update e2e tests to ignore CORS
